libs/wingc/src/jsify.rs

```diff
@@ -1801,31 +1811,53 @@ impl<'a> CaptureScanner<'a> {
 	}
 
 	fn consider_reference(&mut self, path: &[&Symbol]) {
-		if let LookupResult::Found(kind, info) = self.method_env.lookup_nested(path, Some(self.statement_index), 0) {
-			// if the symbol was found in our current enviroment, we move on. that's the essence of capturing:
-			// you reference something that defined in a parent environment.
-			if info.depth == 0 {
-				return;
-			}
+		// look up the reference in the current environment
+		let LookupResult::Found(kind, symbol_info) = self.current_env.lookup_nested(path, self.current_index) else {
+			panic!("unable to find symbol in current environment");
+		};
+
+		// now, we need to determine if the environment this symbol is defined in is a parent of the
+		// method's environment. if it is, we need to capture it.
+		if !(symbol_info.env.is_parent_of(&self.method_env)) {
+			return;
+		}
 
-			let fullname = path.iter().map(|s| s.name.clone()).collect_vec().join(".");
+		let fullname = path.iter().map(|s| s.name.clone()).collect_vec().join(".");
 
-			match kind {
-				SymbolKind::Type(t) => {
-					self.captured_types.insert(fullname, *t);
+		match kind {
+			SymbolKind::Type(t) => {
+				self.captured_types.insert(fullname, *t);
+			}
+			SymbolKind::Variable(var) => {
+				// skip macro functions (like "log" and "assert")
+				if let Type::Function(f) = &*var.type_ {
+					if f.js_override.is_some() {
+						return;
+					}
 				}
-				SymbolKind::Variable(var) => {
-					// skip macro functions (like "log" and "assert")
-					if let Type::Function(f) = &*var.type_ {
-						if f.js_override.is_some() {
+
+				// if the symbol was found in another environment, make sure this scope doesn't have another
+				// variable defined with the same name. if it does, it's an error
+				if !self.current_env.is_same(&symbol_info.env) {
```

Comment:
Isn't this part of the `if` redundant since if this is a capture it was by definition found in another env?